packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -155,19 +169,19 @@ export class WorkerBlockDispatcherService
 
     // eslint-disable-next-line no-constant-condition
     if (true) {
-      /*
-       * Load balancing:
-       * worker1: 1,2,3
-       * worker2: 4,5,6
-       */
-      const workerIdx = this.getNextWorkerIndex();
-      cleanedBlocks.map((height) => this.enqueueBlock(height, workerIdx));
+      let startIndex = 0;
+      while (startIndex < cleanedBlocks.length) {
+        const workerIdx = this.getNextWorkerIndex();
+        const batchSize = Math.min(
+          cleanedBlocks.length - startIndex,
+          await this.maxBatchSize(workerIdx),
+        );
+        cleanedBlocks
+          .slice(startIndex, startIndex + batchSize)
+          .forEach((height) => this.enqueueBlock(height, workerIdx));
+        startIndex += batchSize;
+      }
```

Comment:
Im not sure this makes sense. If we add 5 blocks to worker A, then the 2nd block is really large this will cause issues. 

We should assign blocks to the worker with the most free space I think

Reply:
we can modify `getNextWorkerIndex` to return the worker with most free space